db/migrate/20200617183058_create_ahoy_visits_and_events.rb

```diff
@@ -0,0 +1,61 @@
+class CreateAhoyVisitsAndEvents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :ahoy_visits do |t|
+      t.string :visit_token
+      t.string :visitor_token
+
+      # the rest are recommended but optional
+      # simply remove any you don't want
+
+      # user
+      t.references :user
+
+      # standard
+      t.string :ip
+      t.text :user_agent
+      t.text :referrer
+      t.string :referring_domain
+      t.text :landing_page
+
+      # technology
+      t.string :browser
+      t.string :os
+      t.string :device_type
+
+      # location
+      t.string :country
+      t.string :region
+      t.string :city
+      t.float :latitude
+      t.float :longitude
```

Comment:
Me again on privacy related data. Do we need to track the location down to the coordinates? Right now we tell the users "fill the location field with whatever you want" in https://dev.to/settings/ and I think vagueness is by design. 
Even if we come up with some super clear messaging, users might not know that we're storing location data and might not want to or we should at least give them the option to opt out (I do block GA scripts on purpose for this very reason :D).

Doign a step back: do we need location data? If it's useful for statistics and reports, is country enough? if not, if region enough? if not, is city enough?

Reply:
I think country and/or region is probably fine as far as what is considered PII.

But what if we let signed-in opt in or out of this granularity of tracking? I could see a forem in the future having location-based features which _could_ benefit from location awareness.

Maybe:

Country/region -> Opt in by default (not PII, generally could be helpful for top-level metrics)
Everything more granular -> Opt out by default (maybe just not built in at all until needed/requested, but with this in mind when needed)

City is probably also technically fine but I don't think we have a real need for that and should probably not store it.

Also of note: We pass all our traffic through a Fastly shield so we actually need to reach for `request.env["HTTP_FASTLY_CLIENT_IP"]` instead of the default which might through all of this.

Of course future Forems will not have the same Fastly config so we'll probably need an abstraction for getting this right.